File: php/src/vaas/Options/ForFileOptions.php
<?php

namespace VaasSdk\Options;

use Amp\File\FilesystemDriver;

class ForFileOptions
{
    public const DEFAULT_TIMEOUT = 300;
    public const DEFAULT_REQUEST_ID = null;
    
    public function __construct(
        public bool        $useCache = true,
        public bool        $useHashLookup = true,
        public int         $timeout = self::DEFAULT_TIMEOUT,
		public ?FilesystemDriver $filesystemDriver = null,
        public ?string     $vaasRequestId = self::DEFAULT_REQUEST_ID) {}

    public static function fromVaasOptions(VaasOptions $options): self
    {
        return new self(
            useCache: $options->useCache,
            useHashLookup: $options->useHashLookup,
            timeout: $options->timeout,
            vaasRequestId: self::DEFAULT_REQUEST_ID
        );
    }
}

File: python/src/vaas/options/for_file_options.py
import uuid

class ForFileOptions:
    def __init__(
            self,
            use_hash_lookup = True,
            use_cache = True,
            vaas_request_id = None
    ):
        self.vaas_request_id = vaas_request_id
        self.use_hash_lookup = use_hash_lookup
        self.use_cache = use_cache

    def from_vaas_config(self, vaas_options):
        self.use_hash_lookup = vaas_options.use_hash_lookup
        self.use_cache = vaas_options.use_cache
        return ForFileOptions()
